x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/CrossClusterAccessSubjectInfo.java

```diff
@@ -69,6 +89,15 @@ public Authentication getAuthentication() {
         return authentication;
     }
 
+    public CrossClusterAccessSubjectInfo cleanWithValidation() {
```

Comment:
I've moved the validation logic that used to sit in the service class here for a cleaner encapsulation of cleaning and validating an inbound subject info instance. This, for example, allows us to first check the subject type, then clean, then validate the clean version without exposing all these internal steps and their dependencies beyond this class. In the end, all that callers will see is that they got a clean and valid version or a validation error.